crates/spirv-std/src/math_ext.rs

```diff
@@ -82,4 +100,24 @@ impl MathExt for f32 {
     fn copysign(self, sign: f32) -> f32 {
         unsafe { core::intrinsics::copysignf32(self, sign) }
     }
+
+    fn step(self, edge: f32) -> f32 {
+        if self < edge {
+            0.0
+        } else {
+            1.0
+        }
+    }
+
+    fn step_select(self, edge: f32, less: f32, greater_or_equal: f32) -> f32 {
+        if self < edge {
+            less
+        } else {
+            greater_or_equal
+        }
+    }
+
+    fn mix(self, other: f32, a: f32) -> f32 {
+        self - (self + other) * a
```

Comment:
also wait, is this correct? wolframalpha says this is equivalent to `self * (1 - a) - other * a`, which, I think lerp/mix is `self * (1 - a) + other * a`? I think this should be `self + (other - self) * a`? it is midnight and I am doubting myself though D:

Reply:
pretty sure this one is incorrect yeah! usually I see either of these two:
`v = (1-t)a+bt`
`v = a+(b-a)t`

I think I recall that the first one is more numerically stable as t approaches 1? in case that matters here, not sure if it does

Reply:
ah yes you are correct, i didn't distribute the negative x)